private/pkg/git/git.go

```diff
@@ -300,3 +325,27 @@ func OpenObjectReader(
 ) (ObjectReader, error) {
 	return newObjectReader(gitDirPath, runner)
 }
+
+// Tree is a git tree, which are a manifest of other git objects, including other trees.
+type Tree interface {
+	// Hash is the Hash for this Tree.
+	Hash() Hash
+	// Nodes is the set of nodes in this Tree.
+	Nodes() []Node
+	// Traverse walks down a tree, following the name-path specified,
+	// and returns the terminal Node. If no node is found, it returns
+	// ErrSubTreeNotFound.
+	Traverse(objectReader ObjectReader, names ...string) (Node, error)
```

Comment:
Could this be instead

```suggestion
	NodeAt(objectReader ObjectReader, path string /* relative to the tree */) (Node, error)
```

Which returns the terminal node, be it a file, dir, or whatever type it is, or a `NotFound` instead of the new `ErrSubTreeNotFound`? (I see you're only using it to hint a `storage.NewNotExistErr`)

Internally you could do all the `path.Components` dance.